addons/web/models/properties_base_definition.py

```diff
@@ -0,0 +1,21 @@
+from odoo import _, api, models
+from odoo.exceptions import AccessError
+
+
+class PropertiesBaseDefinition(models.Model):
+    _inherit = "properties.base.definition"
+
+    @api.model
+    def get_properties_base_definition(self, model_name, field_name):
+        """Return the base properties definition if we can read the model."""
+        model = self.env[model_name]
+        model.check_access("read")
+        if model._fields[field_name].type != "properties":
+            raise AccessError(_("You can not read that field definition."))
+        return self.sudo().web_search_read(
```

Comment:
Continuing: if we add a "find or create" argument in _get_record_id_for_properties we can simplify here and rely on the base API, which means the method becomes a simple access check before delegating :)